### ISSUE TITLE
Error: expected number NaN after dispose 3D Column chart and recreate new line chart (Chrome, Edge)

### ISSUE THREAD

**Issue by lticadler** [amcharts/amcharts4#2701]
**Bug description**
Seems like something is stored in amcharts even after disposing a chart.

**Precondition**
- XYChart with 3D column series and XYChartScrollbar.
- legend in separate div (don't know if that is important for the issue)

No errors so far.

My use case is to change the chart into a line chart. (in my app the user could change the layout dynamically)
So the container divs I created for the chart / legend will be reused.

**Steps**
- dispose legend (because of the separate div I found out that legend.parent has to be disposed)
```
if (legend && legend.parent && !legend.parent.isDisposed())
        legend.parent.dispose();
if (legend && !legend.isDisposed())
        legend.dispose();
```
- legendDiv.innerHTML = ''
- dispose chart
```
if (!chart.isDisposed())
        chart.dispose();
this.chart = null; // clear reference
```

No errors so far.

Then I create the new XYChart with line series (from scratch).
As soon as the new chart is rendered (it's rendered correctly and is able to interact as expected) the error is written to the console:
`AMElement.ts:680 Error: <path> attribute d: Expected number, "…0000  L528.2143,NaN`  
Doesn't matter if I add a delay before the creation (see stacks at the bottom of this description).

If I start directly with the line chart - no errors.
If I change from the line chart to the 3D column chart - no errors.
Every time I change from the 3D column chart to line chart the error appears twice.

**Environment (if applicable)**
 - amCharts version: 4.9.27 (but happens also with 4.9.16)
 - Browser and its version: Chrome, Edge (Firefox shows no errors)

**Stacks**
Create new chart directly after dispose
```
Error: <path> attribute d: Expected number, "… L0,-100000 L0,NaN L0,NaN L0,…".
(anonymous function)      	AMElement.ts:680
(anonymous function)      	Object.ts:120
l                         	Object.ts:105
Qkdp.e.each               	Object.ts:119
t.attr                    	AMElement.ts:671
e.setPath                 	Sprite.ts:7797
set                       	Sprite.ts:7777
e.drawSegment             	LineSeriesSegment.ts:151
e.drawSegment             	LineSeries.ts:671
e.closeSegment            	LineSeries.ts:650
e.openSegment             	LineSeries.ts:589
e.openSegmentWrapper      	LineSeries.ts:491
e.validate                	LineSeries.ts:406
(anonymous function)      	ValueAxis.ts:1356
e.each                    	Array.ts:95
t.each                    	List.ts:915
e.getMinMax               	ValueAxis.ts:1355
e.handleExtremesChange    	ValueAxis.ts:2108
e.handleExtremesChange    	DateAxis.ts:2302
(anonymous function)      	EventDispatcher.ts:444
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
e.dispatchImmediately     	Base.ts:1380
e.processValues           	XYSeries.ts:1752
e.validateDataItems       	Series.ts:633
e.validateDataItems       	XYSeries.ts:1351
t.update                  	System.ts:218
(anonymous function)      	System.ts:405
Async call from requestAnimationFrame
u                         	AsyncPending.ts:40
t.requestFrame            	System.ts:404
e._start                  	Animation.ts:448
e.start                   	Animation.ts:457
e.animate                 	Sprite.ts:3198
e.transitTo               	Sprite.ts:3342
e.setState                	Sprite.ts:3265
e.setState                	Container.ts:2023
e.hideReal                	Sprite.ts:8434
e.hide                    	Sprite.ts:8395
e.hideInitially           	Sprite.ts:9441
(anonymous function)      	EventDispatcher.ts:489
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
t.dispatchImmediately     	Registry.ts:289
t.update                  	System.ts:125
(anonymous function)      	System.ts:405
Async call from requestAnimationFrame
u                         	AsyncPending.ts:40
t.requestFrame            	System.ts:404
e.invalidateLayout        	Container.ts:470
e.handleChildRemoved      	Container.ts:438
(anonymous function)      	EventDispatcher.ts:444
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
t.removeIndex             	List.ts:691
t.removeValue             	List.ts:437
set                       	Sprite.ts:1766
e.dispose                 	Sprite.ts:1627
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
e.dispose                 	Component.ts:1098
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
```

Create new chart 5s later per setTimeout after dispose
```
Error: <path> attribute d: Expected number, "… L0,-100000 L0,NaN L0,NaN L0,…".
(anonymous function)      	AMElement.ts:680
(anonymous function)      	Object.ts:120
l                         	Object.ts:105
Qkdp.e.each               	Object.ts:119
t.attr                    	AMElement.ts:671
e.setPath                 	Sprite.ts:7797
set                       	Sprite.ts:7777
e.drawSegment             	LineSeriesSegment.ts:151
e.drawSegment             	LineSeries.ts:671
e.closeSegment            	LineSeries.ts:650
e.openSegment             	LineSeries.ts:589
e.openSegmentWrapper      	LineSeries.ts:491
e.validate                	LineSeries.ts:406
(anonymous function)      	ValueAxis.ts:1356
e.each                    	Array.ts:95
t.each                    	List.ts:915
e.getMinMax               	ValueAxis.ts:1355
e.handleExtremesChange    	ValueAxis.ts:2108
e.handleExtremesChange    	DateAxis.ts:2302
(anonymous function)      	EventDispatcher.ts:444
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
e.dispatchImmediately     	Base.ts:1380
e.processValues           	XYSeries.ts:1752
e.validateDataItems       	Series.ts:633
e.validateDataItems       	XYSeries.ts:1351
t.update                  	System.ts:218
(anonymous function)      	System.ts:405
Async call from requestAnimationFrame
u                         	AsyncPending.ts:40
t.requestFrame            	System.ts:404
e._start                  	Animation.ts:448
e.start                   	Animation.ts:457
e.animate                 	Sprite.ts:3198
e.transitTo               	Sprite.ts:3342
e.setState                	Sprite.ts:3265
e.setState                	Container.ts:2023
e.hideReal                	Sprite.ts:8434
e.hide                    	Sprite.ts:8395
e.hideInitially           	Sprite.ts:9441
(anonymous function)      	EventDispatcher.ts:489
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
t.dispatchImmediately     	Registry.ts:289
t.update                  	System.ts:125
(anonymous function)      	System.ts:405
Async call from requestAnimationFrame
u                         	AsyncPending.ts:40
t.requestFrame            	System.ts:404
e.invalidate              	Sprite.ts:1165
e.invalidate              	Container.ts:483
e                         	Sprite.ts:1046
e                         	Container.ts:347
be                        	Instance.ts:70
Te                        	Instance.ts:388
```

Second error when chart is created directly after dispose (differs minimal to the first error [lines 15-21])
```
Error: <path> attribute d: Expected number, "…0000 L531.0714,NaN L531.0714,N…".
(anonymous function)      	AMElement.ts:680
(anonymous function)      	Object.ts:120
l                         	Object.ts:105
Qkdp.e.each               	Object.ts:119
t.attr                    	AMElement.ts:671
e.setPath                 	Sprite.ts:7797
set                       	Sprite.ts:7777
e.drawSegment             	LineSeriesSegment.ts:151
e.drawSegment             	LineSeries.ts:671
e.closeSegment            	LineSeries.ts:650
e.openSegment             	LineSeries.ts:589
e.openSegmentWrapper      	LineSeries.ts:491
e.validate                	LineSeries.ts:406
e._systemUpdate           	Sprite.ts:9541
e._systemUpdate           	Container.ts:2095
(anonymous function)      	System.ts:305
(anonymous function)      	Object.ts:120
l                         	Object.ts:105
Qkdp.e.each               	Object.ts:119
t.update                  	System.ts:274
(anonymous function)      	System.ts:405
Async call from requestAnimationFrame
u                         	AsyncPending.ts:40
t.requestFrame            	System.ts:404
e._start                  	Animation.ts:448
e.start                   	Animation.ts:457
e.animate                 	Sprite.ts:3198
e.transitTo               	Sprite.ts:3342
e.setState                	Sprite.ts:3265
e.setState                	Container.ts:2023
e.hideReal                	Sprite.ts:8434
e.hide                    	Sprite.ts:8395
e.hideInitially           	Sprite.ts:9441
(anonymous function)      	EventDispatcher.ts:489
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
t.dispatchImmediately     	Registry.ts:289
t.update                  	System.ts:125
(anonymous function)      	System.ts:405
Async call from requestAnimationFrame
u                         	AsyncPending.ts:40
t.requestFrame            	System.ts:404
e.invalidateLayout        	Container.ts:470
e.handleChildRemoved      	Container.ts:438
(anonymous function)      	EventDispatcher.ts:444
(anonymous function)      	EventDispatcher.ts:324
e.each                    	Array.ts:95
t._eachListener           	EventDispatcher.ts:294
t.dispatchImmediately     	EventDispatcher.ts:322
t.removeIndex             	List.ts:691
t.removeValue             	List.ts:437
set                       	Sprite.ts:1766
e.dispose                 	Sprite.ts:1627
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
e.dispose                 	Component.ts:1098
e.disposeChildren         	Container.ts:784
e.dispose                 	Container.ts:1977
```

**Comment by zeroin** [comment]
Could I ask for a codepen of this scenario? Thanks!

**Comment by lticadler** [comment]
I'll try it.
At the moment I'm reducing my code step by step to the minimum necessary.
What I determined so far: if I add no scrollbar there is no error.

**Comment by lticadler** [comment]
Found the one statement which caused the error:
`scrollbarSeries.fillOpacity = 0.3;`
I already had a comment there that this could cause an error in chrome with amCharts.
Back then I also created the Issue #2096
The pen I created there still works in Chrome. But it seems that on creating new charts something is still not fully correct.

**Comment by lticadler** [comment]
Next discovery ;)
If I don't set min and max timestamp on xAxis for the newly created line chart the error is gone.
(When changing back to 3d column chart setting min max is no problem.)

**Comment by lticadler** [comment]
Finally I got it in a pen - reduced to the minimum.
Hope this helps :)
https://codepen.io/chrisadler/pen/QWyZNOg?editors=0010

**Comment by zeroin** [comment]
Thanks, it really helped. I found the problem and it will be fixed in next release, I will let you know when it's ready.

**Comment by martynasma** [comment]
Fixed in [4.9.31](https://github.com/amcharts/amcharts4/releases/tag/4.9.31).

## [4.9.31] - 2020-07-21

### Changed
- Global adapters are being deprecated due to performance reasons. They will still work on some properties, but may be turned off at some future version. If you are using `am4core.globalAdapter`, please consider refactoring your code in some other way.
- If `groupInterval` is manually set, `DateAxis` now prepares all intervals. Normally, the chart does not prepare data sets for time intervals longer than the span of dates in the data. This causes problem in case user sets these intervals manually using `groupInterval`. Note, the groupInterval must be set initially, before data parsing in order data to be grouped into all groupIntervals.

### Fixed
- "Year of week" (`YYYY`) was not being calculated correctly by `DateFormatter`.
- Regression plugin was not working correctly in an all-`ValueAxis` scenarios.
- `ColumnSeries` were not working with `dropFromPreviousState`.
- If data was changed on a chart which was zoomed and there were less data items in the new data than there was before, a JS error was triggered.
- Sometimes a chart with `XYChartScrollbar` with series in it could produce a JS error if series has its `fillOpacity > 0`.
- Series in an `XYChartScrollbar` was not being shown if `options.onlyShowOnViewport = true`.
- `zoomToRectangle()` call was resulting in an error if called after the `MapChart` was rotated.
- `CurvedColumnSeries` columns were not being filled with proper color (black was being used instead).
- A chart with multiple series starting/ending not at the same position could result in wrong zoom on `ValueAxis`.
- `DateAxis` with millisecond `baseInterval` sometimes could show one millisecond less than actual data point in its tooltip.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
